src/components/atoms/ScrollToTopButton/index.tsx

```diff
@@ -0,0 +1,37 @@
+import { FC, useState } from "react";
+
+import * as S from "./styles";
+
+const ScrollToTopButton: FC = () => {
+  const [visible, setVisible] = useState(false);
+
+  const toggleVisible = () => {
```

Comment:
Right now this function is being called on every `scroll` event which is a lot, going up and down randomly on our problem index now gives me hundreds of them 🙃 Therefore we can throttle the execution of the function, in a simpler words, sort of delay it on given conditions: https://www.sitepoint.com/throttle-scroll-events/
You can take a look at this article here. ;)

Reply:
Good point. After I did some reading  and testing some functions I chose to add loadash.throttle to our app  :)